python3Packages.mtcnn: init at 0.1.1; python3Packages.retinaface: init at 0.0.17; python3Packages.deepface: init at 0.0.92; python3Packages.tf-keras: fix dependencies

DerDennisOP opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

DerDennisOP committed:
python3Packages.mtcnn: init at 0.1.1

DerDennisOP committed:
python3Packages.mtcnn: init at 0.1.1

DerDennisOP committed:
python3Packages.retinaface: init at 0.0.17

DerDennisOP committed:
python3Packages.retinaface: init at 0.0.17

DerDennisOP committed:
python3Packages.deepface: init at 0.0.92

DerDennisOP committed:
python3Packages.deepface: init at 0.0.92

Sigmanificient commented on pkgs/development/python-modules/deepface/default.nix:
```diff
@@ -0,0 +1,111 @@
+buildPythonPackage rec {
+  pname = "deepface";
+  version = "0.0.92";
+  format = "setuptools";
```
Use `pyproject = true` and `build-system = [ setuptools ];` to avoid legacy build

Sigmanificient commented on pkgs/development/python-modules/deepface/default.nix:
```diff
@@ -0,0 +1,111 @@
+  src = fetchFromGitHub {
+    owner = "serengil";
+    repo = "deepface";
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

Sigmanificient commented on pkgs/development/python-modules/deepface/default.nix:
```diff
@@ -0,0 +1,111 @@
+
+  pythonImportsCheck = [ "deepface" ];
+
+  meta = with lib; {
```
Avoid `meta = with lib;` https://github.com/NixOS/nixpkgs/issues/208242

Sigmanificient commented on pkgs/development/python-modules/deepface/default.nix:
```diff
@@ -0,0 +1,111 @@
+  pythonImportsCheck = [ "deepface" ];
+
+  meta = with lib; {
+    description = "A Lightweight Face Recognition and Facial Attribute Analysis (Age, Gender, Emotion and Race) Library for Python";
```
```suggestion
    description = "Lightweight Face Recognition and Facial Attribute Analysis (Age, Gender, Emotion and Race) Library for Python";
```

Sigmanificient commented on pkgs/development/python-modules/mtcnn/default.nix:
```diff
@@ -0,0 +1,52 @@
+      --replace-fail "tensorflow." ""
+  '';
+
+  nativeCheckInputs = [ nose3 ];
```
We are actively trying to remove nose, and `nose3` is known to not be a good replacement 
- https://github.com/NixOS/nixpkgs/issues/326513#issuecomment-2254632584

Sigmanificient commented on pkgs/development/python-modules/mtcnn/default.nix:
```diff
@@ -0,0 +1,52 @@
+buildPythonPackage {
+  pname = "mtcnn";
+  version = "0.1.1";
+  format = "setuptools";
```
`format` -> `pyproject = true`

Sigmanificient commented on pkgs/development/python-modules/mtcnn/default.nix:
```diff
@@ -0,0 +1,52 @@
+
+  pythonImportsCheck = [ "mtcnn" ];
+
+  meta = with lib; {
```
Try to avoid `with lib;`

Sigmanificient commented on pkgs/development/python-modules/retinaface/default.nix:
```diff
@@ -0,0 +1,70 @@
+  src = fetchFromGitHub {
+    owner = "serengil";
+    repo = "retinaface";
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

Sigmanificient commented on pkgs/development/python-modules/retinaface/default.nix:
```diff
@@ -0,0 +1,70 @@
+buildPythonPackage rec {
+  pname = "retinaface";
+  version = "0.0.17";
+  format = "setuptools";
```
`format` -> `pyproject = true`

Sigmanificient commented on pkgs/development/python-modules/retinaface/default.nix:
```diff
@@ -0,0 +1,70 @@
+
+  checkPhase = ''
+    # Require internet connection
+    rm tests/test_actions.py
```
Try use `disabledTestPaths`, or move to `preCheck`

Sigmanificient commented on pkgs/development/python-modules/retinaface/default.nix:
```diff
@@ -0,0 +1,70 @@
+
+  pythonImportsCheck = [ "retinaface" ];
+
+  meta = with lib; {
```
Avoid usage of `with lib;`

DerDennisOP committed:
python3Packages.tf-keras: fix dependencies

GaetanLepage commented on pkgs/development/python-modules/deepface/default.nix:
```diff
@@ -0,0 +1,119 @@
+      --replace-fail "return path + \"/data/\"" "return path + \"/share/opencv4/haarcascades/\""
+  '';
+
+  propagatedBuildInputs = [
```
```suggestion
  dependencies = [
```

DerDennisOP replied:
![image](https://github.com/user-attachments/assets/f2fc5fff-418b-4a85-9b84-9180a5fe1100)

You are suggesting breaking changes to my code.

GaetanLepage replied:
I seriously doubt that this change is at the origin of your error...

GaetanLepage commented on pkgs/development/python-modules/mtcnn/default.nix:
```diff
@@ -0,0 +1,55 @@
+      --replace-fail "tensorflow." ""
+  '';
+
+  nativeCheckInputs = [ pytestCheckHook ];
```
This should go after `pythonImportsCheck`

GaetanLepage commented on pkgs/development/python-modules/mtcnn/default.nix:
```diff
@@ -0,0 +1,55 @@
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  propagatedBuildInputs = [
```
```suggestion
  dependencies = [
```

GaetanLepage commented on pkgs/development/python-modules/mtcnn/default.nix:
```diff
@@ -0,0 +1,55 @@
+  pname = "mtcnn";
+  version = "0.1.1";
+  pyproject = true;
+  build-system = [ setuptools ];
```
This should go after `postPatch` and before `dependencies`.

GaetanLepage commented on pkgs/development/python-modules/retinaface/default.nix:
```diff
@@ -0,0 +1,71 @@
+  pname = "retinaface";
+  version = "0.0.17";
+  pyproject = true;
+  build-system = [ setuptools ];
```
same.

GaetanLepage commented on pkgs/development/python-modules/retinaface/default.nix:
```diff
@@ -0,0 +1,71 @@
+    "tests/test_expand_face_area.py"
+  ];
+
+  propagatedBuildInputs = [
```
```suggestion
  dependencies = [
```

DerDennisOP commented:
@GaetanLepage @Sigmanificient  why is there no tensorflow.keras? Can you help me on this?

GaetanLepage commented:
> @GaetanLepage @Sigmanificient why is there no tensorflow.keras? Can you help me on this?

I fear that this `tensorflow.keras` module only exists in recent versions of `tensorflow`.
Unfortunately, we have a pretty outdated one (2.13.0).
One option would be to use `tensorflow-bin`...

DerDennisOP commented:
> > @GaetanLepage @Sigmanificient why is there no tensorflow.keras? Can you help me on this?
> 
> I fear that this `tensorflow.keras` module only exists in recent versions of `tensorflow`. Unfortunately, we have a pretty outdated one (2.13.0). One option would be to use `tensorflow-bin`...

I tried, but tensorflow-bin is also basically broken.

![image](https://github.com/user-attachments/assets/6b7b09f1-a30d-4177-b448-05a13fbc7593)

GaetanLepage commented:
> I tried, but tensorflow-bin is also basically broken.

The whole tensorflow situation is quite depressing unfortunately.
The maintenance of this package requires very high skills that I do not have. So it has been stalling for a while now.

DerDennisOP commented:
> > I tried, but tensorflow-bin is also basically broken.
> 
> The whole tensorflow situation is quite depressing unfortunately. The maintenance of this package requires very high skills that I do not have. So it has been stalling for a while now.

![image](https://github.com/user-attachments/assets/2e6cd97c-ea70-434e-aaf0-00cfcd5c65da)

Yeah, not usable atm.

DerDennisOP commented:
Going back to draft, till tensorflow is fixed.

DerDennisOP committed:
Merge branch 'NixOS:master' into comp

DerDennisOP committed:
python3Packages.torchcrepe: init at 0.0.23

DerDennisOP committed:
Merge pull request #46 from DerDennisOP/torchcrepe

python3Packages.torchcrepe: init at 0.0.23

DerDennisOP committed:
python3Packages.parselmouth: init at 0.4.4

DerDennisOP committed:
Merge pull request #47 from DerDennisOP/parselmouth

python3Packages.parselmouth: init at 0.4.4

mweinelt committed:
python312Packages.tensorflow-bin: unbreak

For some reason we have a distutils package now, and that unbreaks it
until tensorflow upstream gets its shit together.

DerDennisOP commented:
Reopening, due to fin in tensorflow-bin #333568

DerDennisOP committed:
Merge branch 'NixOS:master' into comp

DerDennisOP committed:
temporary fixes

DerDennisOP committed:
temporary fixes

DerDennisOP committed:
temporary fixes

DerDennisOP committed:
Merge branch 'comp' into deepface

DerDennisOP committed:
Merge pull request #48 from DerDennisOP/deepface

Deepface

DerDennisOP committed:
Merge pull request #49 from DerDennisOP/deepface

temporary fixes

DerDennisOP committed:
Merge branch 'comp' into deepface

DerDennisOP committed:
Merge branch 'master' into deepface
